packages/pigeon/platform_tests/alternate_language_test_plugin/ios/Classes/AlternateLanguageTestPlugin.m

```diff
@@ -500,4 +508,53 @@ - (void)callFlutterEchoNullableEnum:(nullable AnEnumBox *)AnEnumBoxed
                          }];
 }
 
+- (void)callFlutterSmallApiEchoString:(nonnull NSString *)aString
+                           completion:(nonnull void (^)(NSString *_Nullable,
+                                                        FlutterError *_Nullable))completion {
+  [self.flutterSmallApiOne
+      echoString:aString
+      completion:^(NSString *valueOne, FlutterError *error) {
+        [self.flutterSmallApiTwo
+            echoString:aString
+            completion:^(NSString *valueTwo, FlutterError *error) {
+              if ([valueOne isEqualToString:valueTwo]) {
+                completion(valueTwo, error);
+              } else {
+                completion(
+                    nil,
+                    [FlutterError
+                        errorWithCode:@"Responses do not match"
+                              message:[NSString stringWithFormat:
+                                                    @"%@%@%@%@",
+                                                    @"Multi-instance responses were not matching: ",
+                                                    valueOne, @", ", valueTwo]
+                              details:nil]);
+              }
+            }];
+      }];
+}
+
+@end
+
+@interface AlternateLanguageTestAPIWithSuffix ()
+@end
+
+@implementation AlternateLanguageTestAPIWithSuffix
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar
+                       suffix:(NSString *)suffix {
+  AlternateLanguageTestAPIWithSuffix *api = [[AlternateLanguageTestAPIWithSuffix alloc] init];
```

Comment:
For all the languages, I think this secondary plugin should take the suffix as an argument, and it should use that in its echo implementation, so that you can address the problem you mentioned of not currently actually testing that they are registering their respective handlers.

Reply:
I did actually write a test that confirms there are multiple (dart) implementations with unique suffixes. Since the integration tests with multiple channels pass, and we know that the dart api's are using the correct channels, we also know that the platforms are using them.

Reply:
Ah, clever! I didn't look closely at the unit test. SGTM.